lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp

```diff
@@ -123,8 +123,8 @@ bool DxilAnnotateWithVirtualRegister::runOnModule(llvm::Module &M) {
   }
   unsigned int Major = 0;
   unsigned int Minor = 0;
-  m_DM->GetDxilVersion(Major, Minor);
-  if (Major < 6 || (Major == 6 && Minor <= 4)) {
+  m_DM->GetValidatorVersion(Major, Minor);
+  if (Major < 1 || (Major == 1 && Minor < 4)) {
```

Comment:
If `Major < 1`, the validator version is probably 0.0, which means it's never meant to be validated, and shouldn't be changed to 1.4.  You should use `DXIL::CompareVersions` to compare, as it takes this into account.

```suggestion
  if (DXIL::CompareVersions(Major, Minor, 1, 4) < 0)) {
```

Reply:
Thanks!